packages/mui-base/src/useControllableReducer/useControllableReducer.ts

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
```

Comment:
This hook is tied to the useListbox and can remain in its directory. It's not meant to be used on its own.